### PR TITLE
Fedora36

### PR DESCRIPTION
**Pull request by e-carlin**
@robnagler the commit history is a total mess in this PR but I find it to be extremely useful when working on rpm-codes. When you're done reviewing please let me merge and I'll clean it up before I do.

### CODE REVIEW

File: installers/rpm-code/codes/boost.sh [hunk 1]
```diff
@@ -1,17 +1,28 @@
 #!/bin/bash
 
+_boost_ver=1.79.0
+
+_boost_dir=boost_${_boost_ver//./_}
+
+# boost 1.72.0 has this problem on gcc11
```
Comment (robnagler) [inline installers/rpm-code/codes/boost.sh:7]: remove this comment?

File: installers/rpm-code/codes/common.sh [hunk 2]
```diff
@@ -66,7 +72,8 @@ common_python() {
 
     # Lots of dependencies so we install here to avoid rpm collisions.
     # Slows down builds of pykern, but doesn't affect development.
-    codes_download pykern
+    # no quotes, because RADIA_RUN_BRANCH_PYKERN may be empty (no qualifier)
+    codes_download pykern ${RADIA_RUN_BRANCH_PYKERN:-}
```
Comment (robnagler) [inline installers/rpm-code/codes/common.sh:76]: I don't think RADIA_RUN_BRANCH_PYKERN is used anywhere. If so, delete. This was an early attempt, I think.

File: installers/rpm-code/codes/ml.sh [hunk 3]
```diff
@@ -9,33 +9,31 @@ ml_python_install() {
...
+        'packaging'
+        'protobuf>=3.9.2,<3.20'
+        'setuptools'
+        'six>=1.12.0'
         'termcolor>=1.1.0'
-        'werkzeug>=0.11.15'
+        'typing_extensions>=3.6.6'
+        'wrapt>=1.11.0'
+        'tensorflow-io-gcs-filesystem>=0.23.1'
+        'grpcio>=1.24.3,<2.0' # sys.byteorder == 'little' on our systems
```
Comment (robnagler) [inline installers/rpm-code/codes/ml.sh:32]: sorting is wrong

**Review by robnagler — CHANGES_REQUESTED** [review]

File: installers/rpm-code/codes/ml.sh (hunk 3 above)
Comment (e-carlin) [reply installers/rpm-code/codes/ml.sh]: 9969cf1

File: installers/rpm-code/codes/common.sh (hunk 2 above)
Comment (e-carlin) [reply installers/rpm-code/codes/common.sh]: 9969cf1

File: installers/rpm-code/codes/boost.sh (hunk 1 above)
Comment (e-carlin) [reply installers/rpm-code/codes/boost.sh]: 9969cf1

**Review by robnagler — APPROVED** [review]

**Pull request merged**